uaclient/security.py

```diff
@@ -859,6 +859,17 @@ def prompt_for_affected_packages(
     if unfixed_pkgs:
         print(_format_unfixed_packages_msg(unfixed_pkgs))
 
+    if util.should_reboot():
```

Comment:
@lucasmoura, we are adding a second call-site which does util.should_reboot and messaging when reboot is needed. What do you think about refactoring util.should_reboot to accept an optional message: str param which we can pass in instead of having a [Entitlement separate class method.check_for_reboot_msg](https://github.com/canonical/ubuntu-advantage-client/blob/master/uaclient/entitlements/repo.py#L85)

Reply:
Also, within this conditional if true, I think we also want to call to `cfg.add_notice("", status.MESSAGE_ENABLE_REBOOT_REQUIRED_TMPL.format(operation="fix operation"))`


If we do that we also need a cfg.remove_notice in reboot_cmds.py that'll clean up the reboot required message.
When you peek at the remove_notices in reboot_cmds.py let us know if you think we need to make better use descr_regex to remove any matching reboot message from notices

Reply:
@blackboxsw I agree on the notices approach, since the user may not reboot immediately after running the `ua fix` operation. However, I don't think allowing `should_reboot` to print messages will make the code better.

We would change the current code from:
```python
if util.should_reboot():
    print(
        status.MESSAGE_ENABLE_REBOOT_REQUIRED_TMPL.format(
            operation="fix operation"
        )    
    )    
    print(
        status.MESSAGE_SECURITY_ISSUE_NOT_RESOLVED.format(issue=issue_id)
    )    
    return
```

to:
```python
msg =  status.MESSAGE_ENABLE_REBOOT_REQUIRED_TMPL.format(
            operation="fix operation"
        )
        + "\n"
       + status.MESSAGE_SECURITY_ISSUE_NOT_RESOLVED.format(issue=issue_id)
if util.should_reboot(msg):
    return
```

My point is that I think it is better to make `should_reboot` be as simple as possible and we drive want we want to do outside of it. Even though we have that same behavior on FIPS, I still think it is better to have the output separate from the `should_reboot` function

Reply:
+1 thanks for your suggestion/response here. Agreed.